core/common/src/main/java/alluxio/grpc/GrpcServer.java

```diff
@@ -34,13 +37,23 @@
    * Create a new instance of {@link GrpcServer}.
    *
    * @param server the wrapped server
+   * @param authServer the authentication server
    * @param serverShutdownTimeoutMs server shutdown timeout in milliseconds
    */
-  public GrpcServer(Server server, long serverShutdownTimeoutMs) {
+  public GrpcServer(Server server, AuthenticationServer authServer, long serverShutdownTimeoutMs) {
     mServer = server;
+    mAuthServer = authServer;
     mServerShutdownTimeoutMs = serverShutdownTimeoutMs;
   }
 
+  /**
+   * @return the authentication server associated with this server
+   */
+  @VisibleForTesting
+  public AuthenticationServer getAuthenticationServer() {
```

Comment:
Can't the test just pass in the authentication server?

Reply:
`GrpcServer` is built by `ServerBuilder` and it dynamically creates the auth server. Removing this would mean adding a new method to builder with a slightly more complex code.